### PR TITLE
Create firestore commit model and reference it everywhere

### PR DESCRIPTION
**Pull request by keyonghan**
Help with https://github.com/flutter/flutter/issues/142951

### CODE REVIEW

File: app_dart/lib/src/model/firestore/commit.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  /// Create [Commit] from a Commit Document.
+  static Commit fromDocument({
+    required Document commitDocument,
+  }) {
+    return Commit()
+      ..fields = commitDocument.fields!
+      ..name = commitDocument.name!;
+  }
+
+ /// The timestamp (in milliseconds since the Epoch) of when the commit
```
Comment (yusuf-goog) [inline app_dart/lib/src/model/firestore/commit.dart:32]: nit: alignment

File: app_dart/lib/src/service/firestore.dart [hunk 2]
```diff
@@ -111,38 +112,42 @@ class FirestoreService {
...
-        kTaskCreateTimestampField: Value(integerValue: commit.timestamp!.toString()),
-        kTaskEndTimestampField: Value(integerValue: kTaskDefaultTimestampValue.toString()),
-        kTaskBringupField: Value(booleanValue: target.value.bringup),
-        kTaskNameField: Value(stringValue: target.value.name),
-        kTaskStartTimestampField: Value(integerValue: kTaskDefaultTimestampValue.toString()),
-        kTaskStatusField: Value(stringValue: Task.statusNew),
-        kTaskTestFlakyField: Value(booleanValue: false),
-        kTaskCommitShaField: Value(stringValue: commit.sha),
-      },
+List<firestore.Task> targetsToTaskDocuments(Commit commit, List<Target> targets) {
```
Comment (yusuf-goog) [inline app_dart/lib/src/service/firestore.dart:115]: nit: is there any way to consolidate the naming of TaskDocument and Task? Not a blocker, but the naming could be slightly confusing in the future.

**Review by yusuf-goog — APPROVED** [review]
With a couple minor comments.

File: app_dart/lib/src/service/firestore.dart [hunk 3]
```diff
@@ -111,38 +112,42 @@ class FirestoreService {
...
+  final Iterable<firestore.Task> iterableDocuments = targets.map(
+    (Target target) => firestore.Task.fromDocument(
+      taskDocument: Document(
+        name: '$kDatabase/documents/$kTaskCollectionId/${commit.sha}_${target.value.name}_1',
```
Comment (ricardoamador) [inline app_dart/lib/src/service/firestore.dart:119]: From looking at the other pull requests related to this effort the '_1' signifies the run attempt correct? And if that is accurate could we maybe add a constant for the initial value like ATTEMPT_INITIAL_VALUE?

File: app_dart/lib/src/model/firestore/commit.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/lib/src/model/firestore/commit.dart]: Fixed.

File: app_dart/lib/src/service/firestore.dart (hunk 2 above)
Comment (keyonghan) [reply app_dart/lib/src/service/firestore.dart:116]: Yeah, there will be a clean up of these namings after we fully switched to firestore. Added a todo in https://github.com/flutter/flutter/issues/142951 to note.

File: app_dart/lib/src/service/firestore.dart (hunk 3 above)
Comment (keyonghan) [reply app_dart/lib/src/service/firestore.dart]: Good point. Added.

**Pull request merged**